src/compas_fab/robots/robot.py

```diff
@@ -496,22 +549,30 @@ def get_group_configuration(self, group, full_configuration):
     def merge_group_with_full_configuration(self, group_configuration, full_configuration, group):
         """Returns a robot's full configuration by merging a group's configuration with a full configuration.
 
+        If the configuration of the joints differ, the group configuration takes
+        precedence over the specified full configuration.
+
         Parameters
         ----------
-        group_configuration : :class:`compas_fab.robots.Configuration`
+        group_configuration : :class:`Configuration`
             The configuration for one of the robot's planning groups.
-        full_configuration : :class:`compas_fab.robots.Configuration`
+        full_configuration : :class:`Configuration`
             The configuration for all configurable joints of the robot.
         group : str
-            The name of the group.
+            The name of the planning group.
 
         Returns
         -------
-        :class:`compas_fab.robots.Configuration`
-            A full configuration: with values for all configurable joints.
+        :class:`Configuration`
+            A full configuration with values for all configurable joints.
+
+        Raises
+        ------
+        ValueError
+            If the group configuration or the full configuration has too few or
+            too many joint values given the :attr:`model` or :attr:`semantics` specified.
```

Comment:
Also somewhat confusingly written. I think it's more like ```If the `full_configuration` does not specify positions for all configuration joints, or if the `group_configuration` does not specify positions for all configurable joints of the given group [...]```

Reply:
Note to self: Fixed on local head.